### PR TITLE
14.0 mig auth ldaps

### PR DESCRIPTION
**Pull request by gurneyalex**

### CODE REVIEW

**Review by leemannd — APPROVED** [review]

**Review by i-vyshnevska — APPROVED** [review]

**Review by primes2h — APPROVED** [review]
It works as expected. :+1:

File: auth_ldaps/README.rst [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+   !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
+
+.. |badge1| image:: https://img.shields.io/badge/maturity-Beta-yellow.png
+    :target: https://odoo-community.org/page/development-status
+    :alt: Beta
+.. |badge2| image:: https://img.shields.io/badge/licence-AGPL--3-blue.png
+    :target: http://www.gnu.org/licenses/agpl-3.0-standalone.html
+    :alt: License: AGPL-3
+.. |badge3| image:: https://img.shields.io/badge/github-OCA%2Fserver--auth-lightgray.png?logo=github
+    :target: https://github.com/OCA/server-auth/tree/13.0/auth_ldaps
```
Comment (petrus-v) [inline auth_ldaps/README.rst:17]:
```suggestion
    :target: https://github.com/OCA/server-auth/tree/14.0/auth_ldaps
```

File: auth_ldaps/README.rst [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+    :alt: OCA/server-auth
+.. |badge4| image:: https://img.shields.io/badge/weblate-Translate%20me-F47D42.png
+    :target: https://translation.odoo-community.org/projects/server-auth-13-0/server-auth-13-0-auth_ldaps
+    :alt: Translate me on Weblate
+.. |badge5| image:: https://img.shields.io/badge/runbot-Try%20me-875A7B.png
+    :target: https://runbot.odoo-community.org/runbot/251/13.0
```
Comment (petrus-v) [inline auth_ldaps/README.rst:23]:
```suggestion
    :target: https://runbot.odoo-community.org/runbot/251/14.0
```

File: auth_ldaps/README.rst [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+#. Access Settings / General Settings / LDAP Authentication / LDAP Server
+#. Check the ´Use LDAPS´ flag
+
+Bug Tracker
+===========
+
+Bugs are tracked on `GitHub Issues <https://github.com/OCA/server-auth/issues>`_.
+In case of trouble, please check there if your issue has already been reported.
+If you spotted it first, help us smashing it by providing a detailed and welcomed
+`feedback <https://github.com/OCA/server-auth/issues/new?body=module:%20auth_ldaps%0Aversion:%2013.0%0A%0A**Steps%20to%20reproduce**%0A-%20...%0A%0A**Current%20behavior**%0A%0A**Expected%20behavior**>`_.
```
Comment (petrus-v) [inline auth_ldaps/README.rst:57]:
```suggestion
`feedback <https://github.com/OCA/server-auth/issues/new?body=module:%20auth_ldaps%0Aversion:%2014.0%0A%0A**Steps%20to%20reproduce**%0A-%20...%0A%0A**Current%20behavior**%0A%0A**Expected%20behavior**>`_.
```

File: auth_ldaps/README.rst [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+
+.. image:: https://odoo-community.org/logo.png
+   :alt: Odoo Community Association
+   :target: https://odoo-community.org
+
+OCA, or the Odoo Community Association, is a nonprofit organization whose
+mission is to support the collaborative development of Odoo features and
+promote its widespread use.
+
+This module is part of the `OCA/server-auth <https://github.com/OCA/server-auth/tree/13.0/auth_ldaps>`_ project on GitHub.
```
Comment (petrus-v) [inline auth_ldaps/README.rst:93]:
```suggestion
This module is part of the `OCA/server-auth <https://github.com/OCA/server-auth/tree/14.0/auth_ldaps>`_ project on GitHub.
```

File: auth_ldaps/i18n/auth_ldaps.pot [hunk 5]
```diff
@@ -0,0 +1,29 @@
+# Translation of Odoo Server.
+# This file contains the translation of the following modules:
+# 	* auth_ldaps
+#
+msgid ""
+msgstr ""
+"Project-Id-Version: Odoo Server 13.0\n"
```
Comment (petrus-v) [inline auth_ldaps/i18n/auth_ldaps.pot:7]:
```suggestion
"Project-Id-Version: Odoo Server 14.0\n"
```

File: auth_ldaps/models/res_company_ldap.py [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+            rec["skip_cert_validation"] = ldap.skip_cert_validation or False
+        return res
+
+    def _connect(self, conf):
+        if conf["is_ssl"]:
+            uri = "ldaps://%s:%d" % (conf["ldap_server"], conf["ldap_server_port"])
+            connection = ldap.initialize(uri)
+            if conf["skip_cert_validation"]:
+                connection.set_option(ldap.OPT_X_TLS_REQUIRE_CERT, ldap.OPT_X_TLS_ALLOW)
+            connection.set_option(ldap.OPT_X_TLS_NEWCTX, 0)
```
Comment (petrus-v) [inline auth_ldaps/models/res_company_ldap.py:38]: It was the same on version 13.0 So I won't block on this one, I'm wondering why setting `ldap.OPT_X_TLS_NEWCTX` option if there are no changes in the fresh initialized connection ?

As far I understand [documentation](https://www.python-ldap.org/en/python-ldap-3.3.0/reference/ldap.html?highlight=set_option#tls-options) this is required while changing TLS context, so I would move this setting likes this:
```suggestion
            if conf["skip_cert_validation"]:
                connection.set_option(ldap.OPT_X_TLS_REQUIRE_CERT, ldap.OPT_X_TLS_ALLOW)
                connection.set_option(ldap.OPT_X_TLS_NEWCTX, 0)
```

File: auth_ldaps/static/description/index.html [hunk 7]
```diff
@@ -0,0 +1,443 @@
...
+!! This file is generated by oca-gen-addon-readme !!
+!! changes will be overwritten.                   !!
+!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!! -->
+<p><a class="reference external" href="https://odoo-community.org/page/development-status"><img alt="Beta" src="https://img.shields.io/badge/maturity-Beta-yellow.png" /></a> <a class="reference external" href="http://www.gnu.org/licenses/agpl-3.0-standalone.html"><img alt="License: AGPL-3" src="https://img.shields.io/badge/licence-AGPL--3-blue.png" /></a> <a class="reference external" href="https://github.com/OCA/server-auth/tree/13.0/auth_ldaps"><img alt="OCA/server-auth" src="https://img.shields.io/badge/github-OCA%2Fserver--auth-lightgray.png?logo=github" /></a> <a class="reference external" href="https://translation.odoo-community.org/projects/server-auth-13-0/server-auth-13-0-auth_ldaps"><img alt="Translate me on Weblate" src="https://img.shields.io/badge/weblate-Translate%20me-F47D42.png" /></a> <a class="reference external" href="https://runbot.odoo-community.org/runbot/251/13.0"><img alt="Try me on Runbot" src="https://img.shields.io/badge/runbot-Try%20me-875A7B.png" /></a></p>
```
Comment (petrus-v) [inline auth_ldaps/static/description/index.html:370]: Not sure if this file is generate automaticaly ?

```suggestion
<p><a class="reference external" href="https://odoo-community.org/page/development-status"><img alt="Beta" src="https://img.shields.io/badge/maturity-Beta-yellow.png" /></a> <a class="reference external" href="http://www.gnu.org/licenses/agpl-3.0-standalone.html"><img alt="License: AGPL-3" src="https://img.shields.io/badge/licence-AGPL--3-blue.png" /></a> <a class="reference external" href="https://github.com/OCA/server-auth/tree/14.0/auth_ldaps"><img alt="OCA/server-auth" src="https://img.shields.io/badge/github-OCA%2Fserver--auth-lightgray.png?logo=github" /></a> <a class="reference external" href="https://translation.odoo-community.org/projects/server-auth-14-0/server-auth-14-0-auth_ldaps"><img alt="Translate me on Weblate" src="https://img.shields.io/badge/weblate-Translate%20me-F47D42.png" /></a> <a class="reference external" href="https://runbot.odoo-community.org/runbot/251/14.0"><img alt="Try me on Runbot" src="https://img.shields.io/badge/runbot-Try%20me-875A7B.png" /></a></p>
```

File: auth_ldaps/static/description/index.html [hunk 8]
```diff
@@ -0,0 +1,443 @@
...
+<li>Access Settings / General Settings / LDAP Authentication / LDAP Server</li>
+<li>Check the ´Use LDAPS´ flag</li>
+</ol>
+</div>
+<div class="section" id="bug-tracker">
+<h1><a class="toc-backref" href="#id3">Bug Tracker</a></h1>
+<p>Bugs are tracked on <a class="reference external" href="https://github.com/OCA/server-auth/issues">GitHub Issues</a>.
+In case of trouble, please check there if your issue has already been reported.
+If you spotted it first, help us smashing it by providing a detailed and welcomed
+<a class="reference external" href="https://github.com/OCA/server-auth/issues/new?body=module:%20auth_ldaps%0Aversion:%2013.0%0A%0A**Steps%20to%20reproduce**%0A-%20...%0A%0A**Current%20behavior**%0A%0A**Expected%20behavior**">feedback</a>.</p>
```
Comment (petrus-v) [inline auth_ldaps/static/description/index.html:407]:
```suggestion
<a class="reference external" href="https://github.com/OCA/server-auth/issues/new?body=module:%20auth_ldaps%0Aversion:%2014.0%0A%0A**Steps%20to%20reproduce**%0A-%20...%0A%0A**Current%20behavior**%0A%0A**Expected%20behavior**">feedback</a>.</p>
```

File: auth_ldaps/static/description/index.html [hunk 9]
```diff
@@ -0,0 +1,443 @@
...
+</ul>
+</div>
+<div class="section" id="maintainers">
+<h2><a class="toc-backref" href="#id7">Maintainers</a></h2>
+<p>This module is maintained by the OCA.</p>
+<a class="reference external image-reference" href="https://odoo-community.org"><img alt="Odoo Community Association" src="https://odoo-community.org/logo.png" /></a>
+<p>OCA, or the Odoo Community Association, is a nonprofit organization whose
+mission is to support the collaborative development of Odoo features and
+promote its widespread use.</p>
+<p>This module is part of the <a class="reference external" href="https://github.com/OCA/server-auth/tree/13.0/auth_ldaps">OCA/server-auth</a> project on GitHub.</p>
```
Comment (petrus-v) [inline auth_ldaps/static/description/index.html:437]:
```suggestion
<p>This module is part of the <a class="reference external" href="https://github.com/OCA/server-auth/tree/14.0/auth_ldaps">OCA/server-auth</a> project on GitHub.</p>
```

**Pull request merged**